Task 3/Task 3.2/Task3.2.1.Library/Enumerators/CycledDynamicArrayEnum.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task3._2._1.Library.Enumerators
+{
+    public class CycledDynamicArrayEnum<T> : DynamicArrayEnum<T>, IEnumerator<T>
+    {
+        public CycledDynamicArrayEnum(CycledDynamicArray<T> array) : base(array) { }
+
+        public new bool MoveNext()
```

Comment:
why `new`? it hides inherited MoveNext() method, but don't overrides it. 

You should use virtual method here